### PR TITLE
test: add testcases for Button Component

### LINKED ISSUE #9: Implement Unit Tests for Button Component
This issue is focused on implementing unit tests for the Button component within our project. Unit testing is a crucial aspect of software development that helps ensure the reliability and correctness of individual code units, such as components.

**Goals:**

- Create comprehensive unit tests for the Button component to validate its functionality.
- Verify that the Button component behaves as expected in various scenarios.
- Ensure that any changes or updates to the Button component in the future do not introduce regressions.

**Tasks:**

1. Write unit tests that cover different aspects of the Button component (e.g., rendering, interaction, event handling).
2. Include test cases for different Button states (e.g., normal, disabled, active).
3. Validate the accessibility of the Button component, ensuring it meets accessibility standards.
4. Ensure that the tests provide sufficient code coverage for the Button component.

**Guidelines:**

- Follow best practices for unit testing.
- Write clear and descriptive test cases that cover both typical and edge cases.
- Pay attention to code quality, readability, and maintainability when writing tests.

### PR DESCRIPTION
**Pull request by coldrohak**
Wrote the unit tests for button component inside a new file index.spec.tsx
Closes #9

### CODE REVIEW

**Comment by coldrohak** [comment]
Hey @rubemfsv, please review it once

File: src/presentation/components/Button/index.spec.tsx
Comment (rubemfsv) [inline src/presentation/components/Button/index.spec.tsx]: use faker here

File: src/presentation/components/Button/index.spec.tsx
Comment (rubemfsv) [inline src/presentation/components/Button/index.spec.tsx]: use faker here

File: src/presentation/components/Button/index.spec.tsx
Comment (rubemfsv) [inline src/presentation/components/Button/index.spec.tsx]: use faker here and below

File: src/presentation/components/Button/index.spec.tsx
Comment (rubemfsv) [inline src/presentation/components/Button/index.spec.tsx:42]: use faker here and below

File: src/presentation/components/Button/index.spec.tsx
Comment (rubemfsv) [inline src/presentation/components/Button/index.spec.tsx]: use faker to handle these data

File: src/presentation/components/Button/index.tsx
Comment (rubemfsv) [inline src/presentation/components/Button/index.tsx]: export in line 5, like export interface...

**Review by rubemfsv — CHANGES_REQUESTED** [review]

**Review by rubemfsv — DISMISSED** [review]

**Comment by coldrohak** [comment]
Hey @rubemfsv , sry i didnt knew about dismiss review actually the tests after your approval were failing here so i pushed a new commit, Can you please guide me if something needs to be corrected?

**Review by rubemfsv — APPROVED** [review]

**Comment by coldrohak** [comment]
@rubemfsv , all of tests are passing on my device but why are they failing on github, please help I am not able to figure out...failing test->CompareFieldsValidation
image

**Pull request merged**
